src/components/Panel.ts

```diff
@@ -1,62 +1,61 @@
+/**
+ * Copyright (c) 2015 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+
 import Viewport from "./Viewport";
 import { OriginalStyle, FlickingPanel, ElementLike } from "../types";
 import { DEFAULT_PANEL_CSS, EVENTS } from "../consts";
 import { addClass, applyCSS, parseArithmeticExpression, parseElement, getProgress } from "../utils";
 
 class Panel implements FlickingPanel {
+  public viewport: Viewport;
   public prevSibling: Panel | null;
   public nextSibling: Panel | null;
 
-  private element: HTMLElement;
-  private viewport: Viewport;
-  private state: {
+  protected state: {
     index: number;
     position: number;
     relativeAnchorPosition: number;
     size: number;
     isClone: boolean;
+    isVirtual: boolean;
     // Index of cloned panel, zero-based integer(original: -1, cloned: [0, 1, 2, ...])
     // if cloneIndex is 0, that means it's first cloned panel of original panel
     cloneIndex: number;
     originalStyle: OriginalStyle;
-    clonedPanels: Panel[];
     cachedBbox: ClientRect | null;
   };
+  private element: HTMLElement;
   private original?: Panel;
+  private clonedPanels: Panel[];
```

Comment:
@WoodNeck Why does clonedPanels come out from state?

Reply:
I thought it was inappropriate to store clonedPanels in state, as I thought `state` is to store some values to be save / restored in getStatus() / setStatus() (saving panel infos is not implemented though)